### PR TITLE
[PEC-1] Remove Jira flag from entries not requiring one

### PR DESCRIPTION
**Pull request by aisayo**
https://ombulabs.atlassian.net/browse/PEC-1

Remove the Jira ticket flag from entries that can be easily identified as not needing one by adding .has_keywords? to DescriptionRules to validate entries containing any characters listed in KEYWORDS

I will abide by the [code of conduct](code_of_conduct.md).

### CODE REVIEW

File: app/domain/time_entry/description_rules.rb
Comment (kindoflew) [inline app/domain/time_entry/description_rules.rb:4]: should `email/slack` be separated?

also, I don't know the features of `Regexp.union` in `ruby`, but can we include additional options here such as: `email(s?)`?

Comment (aisayo) [reply app/domain/time_entry/description_rules.rb:4]: The reason I included email/slack is because some entries in Noko include that entry so the KEYWORDS is a collection of commonly used words for entries. I can also add more if you have suggestions. I did test out if e(-?)mail(s?) will work and it doesn't

Comment (kindoflew) [reply app/domain/time_entry/description_rules.rb:4]: oh ok, cool! 

i normally list them all out separately (like my typical morning entry is `"morning stuff: email, slack, jira, github"`).

according to the docs (https://ruby-doc.org/core-2.4.1/Regexp.html#method-c-union) we should be able to include more complex matches if we pass actual `Regexp` objects, instead of strings. but that might be more work than it's worth.

Comment (aisayo) [reply app/domain/time_entry/description_rules.rb:4]: Thank you for the suggestions! I will add them to the collection of keywords.

I think for now this using strings is a good solution for this scope. We might need to eventually reconsider how the entries are being validated with more intuitive solutions later down the road so it might not be worth it.

**Comment by aisayo** [comment]
@bronzdoc @kindoflew Hi! Just waiting for another review on this PR so we can deploy to staging for QA

**Review by kindoflew — APPROVED** [review]
LGTM!

**Review by mateusdeap — APPROVED** [review]
Seems just fine to me too.

**Review by bronzdoc — APPROVED** [review]

**Comment by bronzdoc** [comment]
@aisayo what are the steps to QA this?

**Pull request merged**
